### DISCUSSION TITLE
Get Prefix Through Vlan Using API

### DISCUSSION THREAD

**Discussion by remchuk** [netbox-community/netbox#13193] [Q&A] [answered]
Hey! 

So Im trying to create a python automation to pull the Vlan of one site and the prefix of the vlan,
I though it is possible to pull it from the vlan but it is not there,

I though about pulling all the information from the Prefixes but there are vlans which doesn't have prefixes so it is problematic.

Is there a way to do it without complicating it ?

**Comment by candlerb** [comment]
What are you using: REST API direct? REST API via Pynetbox? Django ORM (e.g. from custom scripts or reports)? Webhook? Something else?

From the Django ORM (which you can examine interactively using [nbshell](https://docs.netbox.dev/en/stable/administration/netbox-shell/)) there's a relation:

```
>>> v = VLAN.objects.get(id=1)
>>> v.prefixes
<django.db.models.fields.related_descriptors.create_reverse_many_to_one_manager.<locals>.RelatedManager object at 0x7f55ac2812e0>
>>> v.prefixes.all()
<PrefixQuerySet [<Prefix: 10.12.254.0/24>]>
```

From the REST API, you can query `/api/ipam/prefixes/?vlan_id=1`

Note that in general, a single VLAN could have multiple prefixes: e.g. a v4 and a v6 prefix, or a primary and secondary v4 prefix. I observe that `/api/ipam/vlans/1/` includes `prefix_count` but not the actual prefixes.

**Reply by remchuk** [reply]
Im making a get request to the api/ipam/vlans but there is no prefix inside.

And I go to /api/ipam/prefixes because there are some vlans without prefixes and I want to avoid making two get requests.

Is there a way to sent the actual prefix and not the count?

**Reply by candlerb** [reply, marked as answer]
If what you have is the vlan ID, and what you want is the prefix(es) associated with that VLAN, then you need to make an extra request. You can avoid making that extra request if the VLAN shows `"prefix_count": 0` of course.

You might be able to get what you want in a single request using the GraphQL API instead of the REST API.

**Reply by remchuk** [reply]
is there a reason why there is no prefixes associated with the vlans in an API request?

**Reply by remchuk** [reply]
Can I make a feature request? or Is there a valid reason for this feature no to exist? @candlerb

**Reply by candlerb** [reply]
Nothing stops you making a feature request, and I can't speak on behalf of the developers, but my guess is that it's unlikely to be accepted.

REST APIs are inherently limited: they do not have full "SQL join" functionality, and it's normal to have to make multiple queries. It sounds to me like your use case is supported just fine via the REST API as it stands, without having to add special-case functionality. Also as I said, there is GraphQL if you want something more sophisticated.

**Reply by gdmjoho** [reply]
Hi!
Im running API request from ansible. 

If I'm putting a GET request towards URL: "https://netbox.domain.se/api/ipam/prefixes/?vlan_vid=1" I get the prefix, the IDs and everything I need to be able to format my put request later on using the ansible.netbox.netbox module. So this works for me :)
